NitroxServer/GameLogic/EventTriggerer.cs

```diff
@@ -1,41 +1,46 @@
 ﻿using System;
-using System.Timers;
 using NitroxModel.Logger;
 using NitroxModel.Packets;
 
 namespace NitroxServer.GameLogic
 {
     public class EventTriggerer
     {
+
         PlayerManager playerManager;
-        public EventTriggerer(PlayerManager playerManager)
+        EventData eventData;
+
+        public EventTriggerer(PlayerManager playerManager, EventData eventData)
         {
             this.playerManager = playerManager;
-            SetupEventTimers();
+            this.eventData = eventData;
+            if (!eventData.HasTimers())
+            {
+                SetupEventTimers();
+            }
+            eventData.StartTimers();
         }
 
         public void SetupEventTimers()
         {
             // eventually this should be on a better timer so it can be saved, paused, etc
             Log.Debug("Event Triggerer started!");
             double auroraTimer = RandomNumber(2.3d, 4d) * 1200d * 1000d; //Time.deltaTime returns seconds so we need to multiply 1000
-            CreateTimer(auroraTimer * 0.2d, StoryEventType.PDA, "Story_AuroraWarning1");
-            CreateTimer(auroraTimer * 0.5d, StoryEventType.PDA, "Story_AuroraWarning2");
-            CreateTimer(auroraTimer * 0.8d, StoryEventType.PDA, "Story_AuroraWarning3");
-            CreateTimer(auroraTimer, StoryEventType.PDA, "Story_AuroraWarning4");
-            CreateTimer(auroraTimer + 24000, StoryEventType.Extra, "Story_AuroraExplosion");
+            eventData.AddTimer(CreateTimer(auroraTimer * 0.2d, StoryEventType.PDA, "Story_AuroraWarning1"));
+            eventData.AddTimer(CreateTimer(auroraTimer * 0.5d, StoryEventType.PDA, "Story_AuroraWarning2"));
+            eventData.AddTimer(CreateTimer(auroraTimer * 0.8d, StoryEventType.PDA, "Story_AuroraWarning3"));
+            eventData.AddTimer(CreateTimer(auroraTimer, StoryEventType.PDA, "Story_AuroraWarning4"));
+            eventData.AddTimer(CreateTimer(auroraTimer + 24000, StoryEventType.Extra, "Story_AuroraExplosion"));
         }
 
-        public Timer CreateTimer(double time, StoryEventType eventType, string key)
+        public NitroxTimer CreateTimer(double time, StoryEventType eventType, string key)
```

Comment:
Might be clean to make this a static method factory on NitroxTimer.... i.e. `NitroxTimer.From(x, y, z)`

Reply:
Is there a need for this? Usually you would only do static method factory if you default values. Other than the reset value which isnt strickly needed anyway. Unless you are suggesting that we have a method for each of the event timers?